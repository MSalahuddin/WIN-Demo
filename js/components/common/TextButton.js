import React from 'react';
import { TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import Text, { SIZE } from './Text';
import { hitSlop } from '../../styles';

function TextButton({ isUnderlined, onPress, label, disabled, ...rest }) {
  return (
    <TouchableOpacity onPress={onPress} testID="button" hitSlop={hitSlop(10, 10)} disabled={disabled}>
      <Text size={SIZE.XXSMALL} isUnderlined={isUnderlined} {...rest}>
        {label}
      </Text>
    </TouchableOpacity>
  );
}

TextButton.defaultProps = {
  disabled: false,
  isUnderlined: true
};

TextButton.propTypes = {
  disabled: PropTypes.bool,
  isUnderlined: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired
};

export default TextButton;
