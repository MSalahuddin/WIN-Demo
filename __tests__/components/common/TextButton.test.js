import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import TextButton from '../../../js/components/common/TextButton';

describe('Component TextButton', () => {
  const onPress = jest.fn();

  test('renders correctly', () => {
    const output = render(<TextButton onPress={onPress} label="" />);
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when button is pressed', () => {
    const { getByTestId } = render(<TextButton onPress={onPress} label="" />);
    fireEvent(getByTestId('button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
