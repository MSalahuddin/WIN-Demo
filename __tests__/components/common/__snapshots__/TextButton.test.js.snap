// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TextButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="button"
>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="MEDIUM"
    isUnderlined={true}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="XXSMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Calibre-Medium",
          "fontSize": 21,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "underline",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
